### ISSUE TITLE
Scipy cannot be installed

### ISSUE THREAD

**Issue by mcdevitts** [pdm-project/pdm#585]
Using pdm version 1.7.2 on windows in a fresh virtual environment, I cannot add scipy.

```
> pdm add scipy
Adding packages to default dependencies: scipy
✔ �� Lock successful
Changes are written to pdm.lock.
Changes are written to pyproject.toml.
⠙ Resolving packages from lockfile...
See <path to log file> for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='scipy', marker=None, extras=(), specifier=<SpecifierSet('~=1.6')>, editable=False), parent=None)]
```

**Comment by mcdevitts** [comment]
Posted this to the wrong repo. Sorry

**Comment by frostming** [comment]
Can you attach the detailed debug log file together with the content of pyproject.toml?

**Comment by mcdevitts** [comment]
pyproject.toml file content:
`
[project]
name = ""
version = ""
description = ""
authors = [
    {name = "", email = ""},
]
dependencies = [
    "numpy~=1.21",
    "scipy~=1.6",
]
requires-python = ">=3.7"
dynamic = ["classifiers"]
license = {text = "MIT"}

[project.urls]
homepage = ""

[build-system]
requires = ["pdm-pep517"]
build-backend = "pdm.pep517.api"

[tool]
[tool.pdm]
`
[pdm-install-resolve-odngiqib.log](https://github.com/pdm-project/pdm/files/6947195/pdm-install-resolve-odngiqib.log)

**Comment by mcdevitts** [comment]
I tried to dig into this a bit more, and I think I've found the crux of the issue. It's not scipy specifically. I'm using a local pypi repository for some packages. I have some very old versions of scipy on the local pypi repository. These were wheels from back when scipy was not pip installable for windows.

When I remove this repo from my pip.ini file, everything works just fine. How does pdm handle multiple pypi servers?
